polynote-server/src/test/scala/polynote/NotebookRunnerSpec.scala

```diff
@@ -0,0 +1,141 @@
+package polynote
+
+import java.nio.file.{Path, Paths}
+import java.time.{Instant, OffsetDateTime, ZoneOffset}
+
+import org.scalamock.scalatest.MockFactory
+import org.scalatest.{FreeSpec, Matchers}
+import polynote.app.Args
+import polynote.config.PolynoteConfig
+import polynote.kernel.environment.PublishResult
+import polynote.kernel.{Kernel, Output, Result}
+import polynote.kernel.interpreter.Interpreter
+import polynote.messages.{CellID, Notebook, NotebookCell, ShortList}
+import polynote.server.AppEnv
+import polynote.server.repository.{NotebookContent, NotebookRepository}
+import polynote.server.repository.format.NotebookFormat
+import polynote.server.repository.format.ipynb.IPythonFormat
+import polynote.server.repository.fs.{FileSystems, NotebookFilesystem}
+import polynote.testing.ZIOSpec
+import polynote.testing.ZIOSpecBase.BaseEnv
+import zio.clock.Clock
+import zio.{ZEnv, ZIO, ZLayer}
+
+class NotebookRunnerSpec extends FreeSpec with ZIOSpec with MockFactory with Matchers with org.scalamock.matchers.Matchers {
+
+  private val config = PolynoteConfig()
+  private val configLayer = ZLayer.succeed(config)
+  private val fs = mock[NotebookFilesystem]
+  private val fsLayer = ZLayer.succeed(FileSystems.local(fs))
+  private val repo = mock[NotebookRepository]
+  private val repoLayer = ZLayer.succeed(repo)
+  private val kernel = mock[Kernel]
+  private val kernelFactory = Kernel.Factory.const(kernel)
+  private val kernelFactoryLayer = ZLayer.succeed(kernelFactory)
+  private val clock = mock[Clock.Service]
+  private val clockLayer = ZLayer.succeed(clock)
+
+  private val exampleNotebook = NotebookContent(
+    List(
+      NotebookCell(CellID(0), "text", "Some text"),
+      NotebookCell(CellID(1), "scala", """println("hi")"""),
+      NotebookCell(CellID(2), "vega", "this shouldn't be looked at, because we can't locally interpret vega") // TODO: when headless vega interpreter is available, change this to verify
+    ), None
+  ).toNotebook("doesn't matter")
```

Comment:
Can you expand this notebook a bit? Maybe with some variables being shared or python scala interop?

I think eventually we will want to have a bunch of test notebooks that we run through so maybe this isn't urgent.

Reply:
No code in this notebook ever gets run, it's all mocks.

Reply:
🤦 

So it looks like the tests are mostly for checking the CLI arguments and setting the path?

Reply:
Yep, it's actually a unit test for a change 😂